k8sutils/statefulset.go

```diff
@@ -47,11 +47,19 @@ type containerParameters struct {
 	PersistenceEnabled           *bool
 }
 
+type sidecarParameters struct {
+	Name            string
+	Image           string
+	ImagePullPolicy corev1.PullPolicy
+	Resources       *corev1.ResourceRequirements
```

Comment:
nit: I would try to stay consistent with the `Sidecar` struct regarding the field order. Might make sense to update `Sidecar` to have `ImagePullPolicy` come after `Image` like here. Same when instantiating structs.